src/Controls/src/BindingSourceGen/BindingSourceGenerator.cs

```diff
@@ -15,55 +15,186 @@ namespace Microsoft.Maui.Controls.BindingSourceGen;
 [Generator(LanguageNames.CSharp)]
 public class BindingSourceGenerator : IIncrementalGenerator
 {
-	public void Initialize(IncrementalGeneratorInitializationContext initializationContext)
+	// TODO:
+	// Diagnostics
+	// Edge cases
+	// Optimizations
+	static int _idCounter = 0;
```

Comment:
I don't think we need a custom counter (see https://github.com/dotnet/aspnetcore/blob/f9c4b98947d606088b29a305bdf1ea641576d879/src/Http/Http.Extensions/gen/RequestDelegateGenerator.cs#L67C98-L67C119)

Reply:
Index there comes from custom functions defined in asp net core:
https://github.com/dotnet/aspnetcore/blob/main/src/Shared/RoslynUtils/IncrementalValuesProviderExtensions.cs

Let's keep counter for now, and figure a better way later.